% ***************************************************
% Abstract
% ***************************************************
% TO PRODUCE A STAND-ALONE PDF OF YOUR ABSTRACT, uncomment this section and the \end{document} at the end of the file by removing the % from the start of each line.

%\documentclass[12pt, a4paper]{memoir}

%\input{LaTexPackages.tex}

%\begin{document}

%\begin{center}
	%\textbf{\large Your title goes here}

	%\textbf{Abstract}

	%Your Name, The University of Queensland, 20??
%\end{center}

% ********* Enter your text below this line: ********
% Start this section on a new page [this template will automatically handle this]. \\

\noindent
Cancer progression and metastasis are driven by aberrant cell-cell communication between spatially adjacent cells. The mechanism of cellular interactions is through individual cells secreting signalling molecules to influence the behaviour of itself of other cells to coordinate higher biological processes. Studies have shown that cancer cells can develop mutations to suppress immune responses through cellular interaction to foster tumorigenesis. By studying cell-cell interactions in cancer, we can systematically understand the behaviours of cancer cells and unravel the complexity of the interaction of the cancer-immune cell in tumour growth and cancer metastasis. Direct ligand-receptor interactions from adjacent tumours and immune cells are crucial in tumour growth and cancer metastasis (i.e. immune checkpoint inhibitor). These interactions are best captured in the spatial context.  Hence, the ability to capture the genes and/or proteins expression of the cells within the spatial context is essential in studying cell-cell communication in cancer. However, there are thousands of recognised cell signalling molecules which are too many to be experimentally validated. Novel spatial transcriptomics and proteomics methods enable the capture of many molecules in the spatial context to make this more feasible. In this light, this thesis will focus on developing computational approaches to study cancer-immune cell communication in the spatial context.  

The past few years have witnessed a rapid growth of the state-of-the-art spatial-omics technologies including spatial transcriptomics and proteomics. Chapter 1 will thoroughly discuss the experimental development of spatial transcriptomic and proteomic technologies. However, the development of computational approaches for analysing this new data has lagged. Limited tools are available for analysing cell-cell interactions in spatial-omics compared to conventional single-cell RNA sequencing data. There is a great potential for leveraging the spatial context in spatial-omics data to study cellular interactions in cancer.

To address the lack of cell-cell interaction approaches for spatial transcriptomics data, I developed STRISH, a computational analysis framework that scans across the whole tissue section for colocalization of ligand-receptor pairs.  Chapter 2 describes the development of STRISH and the approach to identifying the local co-expression of cells in skin cancer tissues measured by RNAscope. The result of STRISH imaging analysis is used as orthogonal validation of the predictive results from scRNA-seq and spatial transcriptomic sequencing analysis.

In addition to spatial transcriptomic, spatial proteomic can add another layer of information to bring a holistic view of cells in their natural context. The current development of spatial proteomic technologies is grouped into two main categories including multiplexed fluorescence imaging and mass spectrometry imaging technologies. In Chapter 3, two representatives of spatial proteomic data for each approach are applied to capture the protein expression from specimens of human skin cancer and colorectal cancer with 6-plex Opal Polaris and 16-plex Hyperion Imaging Mass Cytometry (IMC), respectively. By leveraging the spatial analysis, we demonstrate the capability to study the tumour microenvironment through community detection and co-occurrence analysis. Furthermore, quantitative measurement of cell composition for each community allows us to deduce biologically interpretable features of skin and colorectal cancers.  

The combination of multimodal spatial-omics data can result in a holistic description of the state of cancer for each patient.  In Chapter 4, I developed, MOSAP, an interactive platform that enables integrated cellular interaction analysis from multimodal spatial-omics experiments. While high throughput experiments for spatial transcriptomic and proteomic are extensively developing, a single experimental platform that produces both types of spatial-mic data in the same sample is not yet available. The experiment is often performed by using multiple adjacent tissue sections. MOSAP enables the seamless alignment of  multiple tissue sections as well as interactively data visualisation through Napari. Using high-throughput CosMx data for spatial transcriptomic in skin cancer and IMC data for multiplexed spatial proteomic in colorectal data, I demonstrate the capability of MOSAP to differentiate the cell-cell interaction across conditions or cancer subtypes. 
Finally, in Chapter 5, I summarise the contributions of this thesis on cellular interaction in cancer through spatial-omics data analysis.

% ***************************************************

%\end{document}