% ***************************************************
% PRELIMINARY PAGES
% ***************************************************
% The instructions contained within this part of the thesis template need to be suppressed from the final thesis. There are instructions on how to do this in the MainThesis.tex file.

% To ensure your work is not suppressed with the instructions please add your text only where instructed.


%***Publications included in this thesis***
\section*{Publications included in this thesis}

\begin{instructional}
    \textbf{Co-first authors peer-reviewed papers}
	
	\begin{enumerate}
    \item \cite{tran2022robust} \textbf{Tran M}, Yoon S, Teoh M, Andersen S, Lam PY, Purdue BW, Raghubar A, Hanson SJ, Devitt K, Jones K, Walters S, Monkman J, Kulasinghe A, Tuong ZK, Soyer HP, Frazer IH and Nguyen Q, ``A robust experimental and computational analysis framework at multiple resolutions, modalities and coverages.'' Front. Immunol., 2022.

    \end{enumerate}
    
    \textbf{Non-first author peer-reviewed papers}
    \begin{enumerate}
        \item \cite{naval2022benchmarking} Naval-Sanchez M, Deshpande N, \textbf{Tran M}, Zhang J, Alhomrani M, Alsanie W, Nguyen Q and Nefzger CM ``Benchmarking of ATAC Sequencing Data From BGI’s Low-Cost DNBSEQ-G400 Instrument for Identification of Open and Occupied Chromatin Regions.'' Front. Mol. Biosci (2022).

        \item Ella Thomson, \textbf{Minh Tran}, Gorjana Robevska, Katie Ayers, Jocelyn van der Bergen, Prarthna Gopalakrishnan Bhaskaran, Eric Haan, Silvia Cereghini, Alla Vash-Margita, Miranda Margetts, Alison Hensley, Quan Nguyen, Andrew Sinclair, Peter Koopman, and Emanuele Pelosi, Functional genomics analysis identifies loss of HNF1B function as a cause of Mayer-Rokitansky-Küster-Hauser syndrome, Hum. Mol. Genet. (2022).
    \end{enumerate}
	
\end{instructional}

% ********* Enter your text below this line: ********


% ***************************************************


%***Submitted manuscripts included in this thesis***
\section*{Submitted manuscripts included in this thesis}

\begin{instructional}
	List manuscript/s submitted for publication here. 
% 	As described above for \textbf{Publications included in the thesis}, on the page immediately preceding the chapter that includes the submitted manuscript, in no more than one (1) page, detail your contribution to the authorship if you are not the sole author.\\
    
    % \textbf{Example:}
    \begin{enumerate}

    \item Xiao Tan, Laura F. Grice, Minh Tran, Onkar Mulay, James Monkman, Tony Blick, Tuan Vo, Ana Clara Simões Flórido Almeida, Jarbas da Silva Motta Junior, Karen Fernandes de Moura, Cleber Machado-Souza, Paulo Souza-Fonseca-Guimaraes, Cristina Pellegrino Baena, Lucia de Noronha, Fernanda Simoes Fortes Guimaraes, Hung N. Luu, Tingsheng Drennon, Stephen Williams, Jacob Stern, Cedric Uytingco, Liuliu Pan, Andy Nam, Caroline Cooper, Kirsty Short, Gabrielle T. Belz, Fernando Souza-Fonseca-Guimaraes, Arutha Kulasinghe, Quan Nguyen, A robust platform for integrative spatial multi-omics analysis to map immune responses to SARS-CoV-2 infection in lung tissues, \textit{In Submission, The EMBO Journal}.
    
    \item Laura F. Grice1, Guiyan Ni, Xinnan Jin, Minh Tran, Onkar Mulay, Duy Pham, Emily E. Killingbeck, Mark T. Gregory, Siok Min Teoh, Tuan Vo, Min Zhang, Zachary Reitz, Katharina Devitt, Liuliu Pan, Arutha Kulasinghe, Yung-Ching Kao, Michael Leon, Sarah R. Murphy, Hiromi Sato, Jazmina Gonzalez Cruz, Snehlata Kumari, Hung N. Luu, Sarah E. Warren, H. Peter Soyer, Mitchell S. Stark, Kiarash Khosrotehrani, Ian Frazer, Youngmi Kim, Quan Nguyen, Comparative skin cancer atlas and interactome: A multi\-modal spatial approach to uncovering the cells and interactions underlying skin cancer diversity, \textit{In Submission, Nature Communication}.

    \item Maryam Shojaei, Amir Shamshirian, James Monkman, Laura Grice, Chin Wee Tan, Min Teo, Minh Tran, Gustavo Rodrigues Rossi, Timothy R Mcculloch, Marek Nalos, Maedeh Raei, Alireza Razavi, Roya Ghasemian, Mobina Gheibi, Fatemeh Roozbeh, Peter D. Sly, Kirsten M Spann, Keng Yih Chew, Yanshan Zhu, Yao Xia, Timothy J Wells, Alexandra Cristina Senegaglia, Carmen Lúcia Kuniyoshi Rebelatto, Claudio Luciano Franck, Anna Flavia Ribeiro dos Santos, Lucia de Noronha, Sepideh Motamen, Reza Valadan, Omolbanin Amjadi, Rajan Gogna, Esha Madan, Reza Alizadeh-Navaei, Liliana Lamperti, Felipe A Zuñiga, Estefania Nova-Lamperti, Gonzalo Labarca, Ben Knippenberg, Velma Herwanto, Ya Wang, Amy Phu, Tracy Chew, Timothy Kwan, Karan Kim, Sally Teoh, Tiana M Pelaia, Win Sen Kuan, Yvette Jee, Jon Iredell, Ken O'Byrne, John F Fraser, Melissa J Davis, Gabrielle Belz, Majid Ebrahimi Warkiani, Carlos Salomon, Fernando Guimaraes, Quan Nguyen, Anthony Mclean, Arutha Kulasinghe, Kirsty R. Short and Benjamin Tang, IFI27 transcription is an early predictor for COVID-19 outcomes, a multi-cohort observational study, \textit{In Submission, Frontiers in Immunology}
    \end{enumerate}
\end{instructional}

% ********* Enter your text below this line: ********


% ***************************************************


%***Other publications during candidature***
\section*{Other publications during candidature}

\begin{instructional}
    List other publications arising during your candidature using the standard citation format for your discipline. 
    
    \subsection*{Conference abstracts}
    \begin{enumerate}

    \item Laura Grice, Guiyan Ni, Xinnan Jin, Minh Tran, Emily Killingbeck, Mark Gregory, Onkar Mulay, Siok-Min Teoh, Arutha Kulasinghe, Michael Leon, Sarah Murphy, Sarah Warren, Youngmi Kim, Quan Nguyen, A single-cell, spatial multiomics atlas and cellular interactome of all major skin cancer types, \textit{American Association for Cancer Research Annual Meeting 2022}
    
    \item Minh Tran, Andrew Su, HoJoon Lee, Richard Cruz, Lance Pflieger, Ashely Dean, Quan Nguyen, Hanlee P. Ji, Terence Rhodes, Understanding the tumour immune microenvironment of stage III colorectal cancer using multiplexed imaging mass cytometry, \textit{Royal College of Pathologists of Australasia-Pathology Update}
    
    \item Minh Tran, Andrew Su, HoJoon Lee, Richard Cruz, Lance Pflieger, Ashely Dean, Quan Nguyen, Hanlee Ji and Terence Rhodes, Spatial single-cell analysis of colorectal cancer tumour using multiplexed imaging mass cytometry, \textit{Immuno Therapy of Cancer}

    \item 
    \end{enumerate}

    \subsection*{Book chapters} 

    \begin{enumerate}

    \item \cite{tan2021applying} \textbf{Tan, X.}, Su, A. T., Hajiabadi, H., Tran, M., and Nguyen, Q., Applying machine learning for integration of multi-modal genomics data and imaging data to quantify heterogeneity in tumour tissues, \textit{Artificial Neural Networks}, 209-228 (2021),


    \end{enumerate}

\end{instructional}

% ********* Enter your text below this line: ********
% Tan, Xiao, et al. "Applying machine learning for integration of multi-modal genomics data and imaging data to quantify heterogeneity in tumour tissues." : .
% Tan, X., Su, A. T., Hajiabadi, H., Tran, M., & Nguyen, Q. (2021). Applying machine learning for integration of multi-modal genomics data and imaging data to quantify heterogeneity in tumour tissues. Artificial Neural Networks, 209-228.
% ***************************************************


%***Contributions by others to the thesis***
\section*{Publication included in this thesis}

\begin{instructional}
	Tran $\&$ Yoon et al \cite{tran2022robust}: A robust experimental and computational analysis framework at multiple resolutions, modalities and coverages, Frontiers in Immunology, 2022. 
\begin{table}[ht]	
\begin{tabularx}{\textwidth}{|X|X|} 
 \hline
 Contributor & Statement of contribution \\ 
 \hline\hline
 Minh Tran (Candidate) & Conducted relevant experiments (5\%). \\ 
 & Developed the algorithms and analysed data (60\%)  \\ 
 & Drafting and Production (25\%) \\
 \hline
 Sohye Yoon & Conducted relevant experiments (RNAscope and ddPCR experiments) (50\%)   \\
 & Drafting and Production (25\%) \\
 \hline
 Teoh M, Andersen S, Lam PY, Purdue BW, Raghubar A, Hanson SJ, Jones K, Walters S,  & Conducted relevant experiments (Visium, RNA sequencing, H$\&$E staining) (30\%) \\
 \hline
 Monkman J, Kulasinghe A & Conducted relevant experiments (Polaris data) (10\%) \\
 \hline
 Quan Nguyen & Conceived experiments (95\%) \\
  & Developed the algorithms (40\%) \\
  & Drafting and Production (45\%) \\
 \hline
 Tuong ZK, Frazer IH & Drafting and Production (5\%) \\
 \hline
 Devitt K & Conducted relevant experiments (Single cell RNA sequencing data) (10\%) \\
 \hline
 Soyer HP, Frazer IH & Provided access to patient samples (100\%)  \\ [1ex] 
 \hline
\end{tabularx}
\end{table} 
\end{instructional}

% ********* Enter your text below this line: ********


% ***************************************************


%***Statement of parts of the thesis submitted to qualify for the award of another degree***
\pagebreak
\section*{Statement of parts of the thesis submitted to qualify for the award of another degree}
None

% ********* Enter your text below this line: ********


% ***************************************************


%***Research involving human or animal subjects***
\section*{Research involving human or animal subjects}

\begin{instructional}
	All research involving human or animal subjects requires prior ethical review and approval by an independent review committee. At UQ, the relevant committee for research involving human subjects is the \href{http://www.uq.edu.au/research/integrity-compliance/human-ethics}{\color{blue}{Human Ethics Unit}} and the relevant committee for research involving animal subjects is the relevant \href{http://www.uq.edu.au/research/integrity-compliance/animal-welfare}{\color{blue}{Animal Ethics Committee}}.  Please provide details of any ethics approvals obtained including the ethics approval number and name of approving committees.  A copy of the ethics approval letter must be included in the thesis appendix.\\
    
    \noindent
	If no human or animal subjects were involved in this research please state: ``No animal or human subjects were involved in this research''.
\end{instructional}

% ********* Enter your text below this line: ********


% ***************************************************


%***Acknowledgements***
\clearpage
\section*{Acknowledgments}

\begin{instructional}
    Start this section on a new page [the template will handle this for you].\\
    
    \noindent
    Acknowledgements recognise those who have been instrumental in the completion of the project.  Acknowledgements should include any professional editorial advice received including the name of the editor and a brief description of the service rendered.
\end{instructional}

% ********* Enter your text below this line: ********


% ***************************************************


%***Financial Support***
\clearpage
\section*{Financial support}

\begin{instructional}
    Start this section on a new page [the template will handle this for you].\\
    
    \noindent
    If you are the recipient of an Australian Government Research Training Program (RTP) scholarship, you are required to acknowledge this contribution.  Please include the text below:\\
    
    \noindent
    ``This research was supported by an Australian Government Research Training Program Scholarship''\\
    
    \noindent
    If you received any other financial support for your project, you are also required to acknowledge the funding body/bodies in this section.\\
    
    \noindent
    If no financial provided then state ``No financial support was provided to fund this research''.
\end{instructional}

% ********* Enter your text below this line: ********


% ***************************************************


%***Keywords***
\section*{Keywords}

\begin{instructional}
	Maximum 10 words; use lower case throughout, separating words/phrases with commas. For example: word, word word, word, word, word word
\end{instructional}
% ********* Enter your text below this line: ********


% ***************************************************


%***Australian and New Zealand Standard Research Classifications (ANZSRC)***
\section*{Australian and New Zealand Standard Research Classifications (ANZSRC)}

\begin{instructional}
    Provide data that links your thesis to the disciplines and discipline clusters in the Federal Government’s Excellence in Research for Australia (ERA) initiative.\\
    
    \noindent
    Please allocate the thesis a \textbf{maximum of 3} \href{http://www.abs.gov.au/Ausstats/abs@.nsf/Latestproducts/6BB427AB9696C225CA2574180004463E?opendocument}{\color{blue}{Australian and New Zealand Standard Research Classifications (ANZSRC) codes}} at the \textbf{6 digit level} and include the descriptor and a percent weighting for each code. Total percent must add to 100.\\


% \textbf{Example:}\\


%     ANZSRC code: 060101, Analytical Biochemistry, 60\% \\
%     \indent ANZSRC code: 060104, Cell Metabolism, 20\% \\
%     \indent ANZSRC code: 060199, Biochemistry and Cell Biology not elsewhere classified, 20\%
\end{instructional}

% ********* Enter your text below this line: ********


% ***************************************************


%***Fields of Research (FoR) Classification***
\section*{Fields of Research (FoR) Classification}

% \begin{instructional}
%     Allows for categorisation of the thesis according to the field of research. \\
    
%     \noindent
%     Please allocate the thesis a \textbf{maximum of 3} \href{http://www.abs.gov.au/Ausstats/abs@.nsf/Latestproducts/6BB427AB9696C225CA2574180004463E?opendocument}{\color{blue}{Fields of Research (FoR) Codes}} at the \textbf{4 digit level} and include the descriptor and a percent weighting for each code. Total percent must add to 100. \\

% \textbf{Example:}\\


\indent FoR code: 0601, Bioinformatics, 100\% \\
% \end{instructional}

% ********* Enter your text below this line: ********


% ***************************************************


%***Order of remaining thesis content***
% \begin{instructional}
% \section*{Order for the Remainder of the Thesis}
% \noindent
%     Remainder of the thesis should be in the following order

%     \begin{itemize}
%         \item Dedications (if applicable)
%         \item Table of Contents
%         \item List of Figures and Tables
%         \item List of Abbreviations used in the thesis
%         \item Main text of the thesis
%         \item Bibliography or List of References
%         \item Appendices
%     \end{itemize}

% \noindent
% \textbf{Date of thesis template release:} 22 March 2019
% \end{instructional}
% \clearpage


% %***Dedication***
% %If you wish to add a dedication (if appropriate), do so here.
% %If not, comment out from here...
% 	\rmfamily
% 	\normalfont
    
%     	\begin{vplace}[1]
% 		\begin{center}
%         Dedication (if applicable)
% 		\end{center}
% 	\end{vplace}

% %... to here.

% \clearpage
\pagebreak
\pagestyle{headings}

% \section{Timeline to thesis submission}
% \begin{table}[ht]
% \centering
% \caption{Research and publication plans}
% \begin{tabular}{||P{5cm}|P{0.66cm}|P{0.66cm}|P{0.66cm}|P{0.66cm}|P{0.66cm}||} 
%  \hline
%    \multirow{2}{3cm}{Research plans}& \multicolumn{1}{c|}{2022} & \multicolumn{4}{c|}{2023}   \\ [0.33ex] 
%     \cline{2-6}
%    & Dec & Jan & Feb & Mar &  Apr  \\ 
%  \hline
%  Chapter 1,2: Revision &  \cellcolor{mygreen!50}\checkmark  &   & & &  \\%  Conferences &  & 
%   \hline
%   Chapter 3: Discussion &  \cellcolor{mygreen!50}\checkmark  &   & & &  \\%  Conferences &  & 
%   \hline
%   Chapter 4: Development MOSAP - MultiOmics Spatial Analysis Platform  &  \cellcolor{mygreen!50}\checkmark  &   & & &  \\%  Conferences &  & 
%   \hline
%   Chapter 4:  The variation of tumour heterogeneity across skin cancer subtypes using CosMx dataset  &  \cellcolor{mygreen!50}\checkmark  &   & & &  \\%  Conferences &  & 
%   \hline
%   Chapter 4:  The tumor microenvironment analysis across patients in colorectal cancer samples &  \cellcolor{mygreen!50}  & \cellcolor{mygreen!50}  & & &  \\%  Conferences &  & 
%   \hline
%   Chapter 5:  Conclusion &  \cellcolor{mygreen!50}  & \cellcolor{mygreen!50}  &  & &  \\%  Conferences &  & 
%   \hline
%   Thesis submission and revision &    & \cellcolor{mygreen!50}  & \cellcolor{mygreen!50} & \cellcolor{mygreen!50} & \cellcolor{lightgray!50} \\%  Conferences &  & 
  % \hline
%  Conferences &  & \cellcolor{mygreen!50} \checkmark  & \cellcolor{mygreen!50}\checkmark &  &  & \cellcolor{mygreen!50} & & &\\
%  \hline
%  \textbf{Aim 2.1}: Using spatial proteomic to identify cell types within tissue & \cellcolor{mygreen!50}  & \cellcolor{mygreen!50}  & \cellcolor{mygreen!50}\checkmark  &  &  &  &  &  &\\
%   \hline
% \textbf{Aim 2.2}: Analyse CCC and cell communities with spatial proteomic  &   &  &  & \cellcolor{mygreen!50} & \cellcolor{mygreen!50} & \cellcolor{mygreen!50} &  &  &\\
%   \hline
%   \textbf{Aim 3}: Framework for Integrating of multimodal spatial -omic data and CCC detection within cancer tissue  &   &  &   &  & \cellcolor{mygreen!50} & \cellcolor{mygreen!50} & \cellcolor{mygreen!50} & \cellcolor{mygreen!50} &\\
%   \hline
%     M3 report and interview &   &  &   &  &  &  & \cellcolor{mygreen!50} &  &\\
%   \hline
%   Thesis writing &   &  &   &  &  & \cellcolor{mygreen!50} & \cellcolor{mygreen!50} & \cellcolor{mygreen!50} &\\
%   \hline
%     Reserve &   &  &   &  &  &  &  &  & \cellcolor{lightgray!50} \\
% \hline
%    & \multicolumn{4}{c|}{\cellcolor{mygreen!50}Year 2}  &  \multicolumn{3}{c|}{\cellcolor{mygreen!50}Year 3}   &  \cellcolor{lightgray!50} & \cellcolor{lightgray!50}  \\ 
% \hline
% \end{tabular}
% \label{table:timeline}
% \end{table}
\clearpage
%***Table of Contents***
%These generate the table of contents, list of figures, and list of tables from items tagged with a \label{} command.
\tableofcontents
	\clearpage
\listoffigures
	\clearpage
\listoftables
\newpage
\input{./PreliminaryAndBackPages/Symbols.tex} %List of symbols. REMOVE IF NOT NEEDED.




%***End of front matter***